iceoryx_dust/container/include/iox/detail/fixed_position_container.inl

```diff
@@ -51,6 +52,130 @@ inline FixedPositionContainer<T, CAPACITY>::~FixedPositionContainer() noexcept
     }
 }
 
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(const FixedPositionContainer& rhs) noexcept
+{
+    *this = rhs;
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(FixedPositionContainer&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>&
+FixedPositionContainer<T, CAPACITY>::operator=(const FixedPositionContainer& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        IndexType i = Index::FIRST;
+        ConstIterator rhs_it = rhs.begin();
+
+        // copy using copy assignment util rhs iter becomes invalid
+        for (; rhs_it.to_index() != Index::INVALID; i++, rhs_it++)
+        {
+            if (m_status[i] == SlotStatus::USED)
+            {
+                // clean original data by it's dtor
+                erase(i);
+            }
+            m_data[i] = *rhs_it;
+            m_status[i] = SlotStatus::USED;
+            // can this line be removed?
+            m_next[i] = static_cast<IndexType>(i + 1U);
+        }
+
+        // correct next
+        m_next[i] = Index::INVALID;
+
+        // erase rest USED element in rhs, m_next and m_status will be also modified by erase()
+        while (i < Index::INVALID)
+        {
+            if (m_status[i] == SlotStatus::USED)
+            {
+                break;
+            }
+
+            i++;
+        }
+
+        // if rest USED slot found
+        if (i != Index::INVALID)
+        {
+            for (Iterator lhs_it = this->iter_from_index(i); lhs_it.to_index() != Index::INVALID; lhs_it++)
+            {
+                erase(lhs_it);
+            }
+        }
+
+        // member update
+        // may this cause problem when size == capacity?
+        m_begin_free = static_cast<IndexType>(rhs.size());
+        m_begin_used = Index::FIRST;
+        m_size = rhs.m_size;
+    }
+    return *this;
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>&
+FixedPositionContainer<T, CAPACITY>::operator=(FixedPositionContainer&& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        IndexType i = Index::FIRST;
+        Iterator rhs_it = rhs.begin();
+
+        // copy using copy assignment util rhs iter becomes invalid
+        for (; rhs_it.to_index() != Index::INVALID; i++, rhs_it++)
+        {
+            if (m_status[i] == SlotStatus::USED)
+            {
+                // clean original data by it's dtor
+                erase(i);
+            }
+            m_data[i] = std::move(*rhs_it);
```

Comment:
This must be changed similarly to the copy assignment.

When looking at the code, it is almost identical the copy assignment, except the line with `m_data[i] = std::move(*rhs_it);` and the `rhs.clear()` at the bottom. Can you think of a way how to reduce the code duplication?

Reply:
See https://github.com/eclipse-iceoryx/iceoryx/commit/e28c2d798c936d69fc90e82d37abaf50d6b97017.